feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts

```diff
@@ -1,38 +1,36 @@
-import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import {
   Configurator,
   ConfiguratorCommonsService,
   ConfiguratorGroupsService,
   GenericConfigurator,
 } from '@spartacus/core';
+import { ConfigRouterExtractorService } from '@spartacus/storefront';
 import { Observable } from 'rxjs';
 import { map, switchMap, take } from 'rxjs/operators';
-import { ConfigRouterExtractorService } from '../../generic/service/config-router-extractor.service';
 
 @Component({
   selector: 'cx-config-previous-next-buttons',
   templateUrl: './config-previous-next-buttons.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfigPreviousNextButtonsComponent implements OnInit {
-  configuration$: Observable<Configurator.Configuration>;
+export class ConfigPreviousNextButtonsComponent {
+  configuration$: Observable<
+    Configurator.Configuration
+  > = this.configRouterExtractorService
+    .extractRouterData()
+    .pipe(
+      switchMap((routerData) =>
+        this.configuratorCommonsService.getConfiguration(routerData.owner)
+      )
+    );
 
   constructor(
-    private configuratorGroupsService: ConfiguratorGroupsService,
-    private configuratorCommonsService: ConfiguratorCommonsService,
-    private configRouterExtractorService: ConfigRouterExtractorService
+    protected configuratorGroupsService: ConfiguratorGroupsService,
+    protected configuratorCommonsService: ConfiguratorCommonsService,
+    protected configRouterExtractorService: ConfigRouterExtractorService
   ) {}
 
-  ngOnInit(): void {
-    this.configuration$ = this.configRouterExtractorService
-      .extractRouterData()
-      .pipe(
-        switchMap((routerData) =>
-          this.configuratorCommonsService.getConfiguration(routerData.owner)
-        )
-      );
-  }
-
```

Comment:
At very least we can get rid of either `onPrevious` or `navigateToPreviousGroup` methods, and just have one that will be called on the click event. 
Similar can be done for the `onNext`.

Reply:
ok